### ISSUE TITLE
How to make the number of species surviving the simulation sufficiently high

### ISSUE THREAD

**Issue by Yokelison** [PoisotLab/BioEnergeticFoodWebs.jl#134]
## Expected behavior

-  I would like the number of species surviving at the end (the number of species with biomass > 0 in result[:B]) to be higher after using the simulate function.（used simulate function without any other extra parameters）

## Actual behavior

- The number of surviving species was relatively low, with less than 50% of the food web of more than 80 species surviving after simulation with the simulate function.

## Steps to reproduce

1. foodweb = nichemodel(80, 0.2)
2. foodwebpara = model_parameters(foodweb)
3. foodwebbio =rand(size(foodweb, 1))
4. result=simulate(foodwebpara, foodwebbio)
5. precent = length(findall(x -> x > 0, result_in[:B][end, :]))/length(foodweb[1,:])

## System information

- `julia --version` 1.3.0
- `Pkg.status("BioEnergeticFoodWebs")`
- Windows 11

**Comment by tpoisot** [comment]
You should probably be using https://github.com/BecksLab/EcologicalNetworksDynamics.jl

**Comment by Yokelison** [comment]
OK， i'll try thank you

**Comment by ismael-lajaaiti** [comment]
Hello @Yokelison, as @tpoisot directed you toward our library (thank you for that), I'll answer here. However, I invite you to continue the conversation on our repository by opening a new issue [there](https://github.com/BecksLab/EcologicalNetworksDynamics.jl), if that's good for you. 

First, I see that you have an old Julia version, so I would recommend [upgrading your Julia version](https://julialang.org/downloads/). For your information, our package is tested on Julia v1.10. Plus, I believe that package pre-compilation times are much faster on latest Julia versions, you may appreciate the speed boost. 

Regarding your issue, here is an equivalent code to the one you provided, using the syntax of EcologicalNetworksDynamics

```julia
using EcologicalNetworksDynamics
using Random
Random.seed!(123) # Seed for reproducibility.

S = 80 # Initial number of species.
C = 0.2 # Connectance of trophic links.
fw = Foodweb(:niche; S, C)
model = default_model(fw)
B0 = rand(S)
tmax = 1_000
sol = simulate(model, B0, tmax)
count(sol[end] .> 1e-6) # 66 surviving species.
```

Here I got 66 species out of 80 initial ones, which is not bad. Actually, it is common to observe many species extinctions during community assembly. To increase the number of surviving species, I see two options.

### 1. Tune the parameters

Try to change the model parameters. For instance, by lowering the connectance (`C`), I obtain more than 70 species. However, please **note that this number can vary greatly between simulation runs**, that is, if re-run the same code (with a different seed) I may obtain a very different number of surviving species (see point 2.).

```julia 
C = 0.1 # **Lowered connectance.**
fw = Foodweb(:niche; S, C)
model = default_model(fw)
B0 = rand(S)
tmax = 1_000
sol = simulate(model, B0, tmax)
count(sol[end] .> 1e-6) # 76 surviving species.
```

You can also play with the predator-prey body mass ratio `Z` that modulates the interaction strengths, for example 

```julia
fw = Foodweb(:niche; S, C)
model = default_model(fw, BodyMass(Z = 10))
B0 = rand(S)
tmax = 1_000
sol = simulate(model, B0, tmax)
count(sol[end] .> 1e-6) # 43.. but can be much higher.
```

### 2. Repeat and select communities with more species

You could do a `for` loop, save all communities with their number of surviving species and only keep the ones with the highest number of species.

I hope that answers your question. If not, I invite to open issue on our repository.

**Comment by Yokelison** [comment]
Thank you for you so detailed reply,  I have updated to julia 1.10.0, and used the package you recommended, and yes, using this package perfectly solves the problem of insufficient numbers in the leftovers, and in my explorations I have found that food webs with fewer producers result in insufficient numbers of species remaining at the end. I experimented with actual food webs (actual food webs generally have more producers than food webs generated using nichemodel, which may be a hidden flaw in nichemodel food webs) and the final number of species remaining was substantial. BTW, why are the simulation results of the nichemodel function so different between the two packages? Is the difference between the two functions enough to cause a huge difference in the number of organisms that end up surviving?
![ac34873e9481d93e50442f2c49aa891](https://github.com/user-attachments/assets/3cd58e94-32db-4b67-a4e2-7a29d0e20582)
